packages/orama/src/components/documents-store.ts

```diff
@@ -1,54 +1,64 @@
+import { DocumentID, getInternalDocumentId, InternalDocumentID, InternalDocumentIDStore } from './internal-document-id-store.js';
+import { Document, IDocumentsStore, OpaqueDocumentStore } from '../types.js';
 
 export interface DocumentsStore extends OpaqueDocumentStore {
-  docs: Record<string, Document | undefined>
+  sharedInternalDocumentStore: InternalDocumentIDStore;
+  docs: Record<InternalDocumentID, Document | undefined>,
   count: number
 }
 
 export type DefaultDocumentsStore = IDocumentsStore<DocumentsStore>
 
-export async function create(): Promise<DocumentsStore> {
+export async function create(sharedInternalDocumentStore: InternalDocumentIDStore): Promise<DocumentsStore> {
   return {
+    sharedInternalDocumentStore,
     docs: {},
     count: 0,
   }
 }
 
-export async function get(store: DocumentsStore, id: string): Promise<Document | undefined> {
-  return store.docs[id]
+export async function get(store: DocumentsStore, id: DocumentID): Promise<Document | undefined> {
+  const internalId = getInternalDocumentId(store.sharedInternalDocumentStore, id);
+
+  return store.docs[internalId]
 }
 
-export async function getMultiple(store: DocumentsStore, ids: string[]): Promise<(Document | undefined)[]> {
+export async function getMultiple(store: DocumentsStore, ids: DocumentID[]): Promise<(Document | undefined)[]> {
   const found: (Document | undefined)[] = Array.from({ length: ids.length })
 
   for (let i = 0; i < ids.length; i++) {
-    found[i] = store.docs[ids[i]]
+    const internalId = getInternalDocumentId(store.sharedInternalDocumentStore, ids[i])
+    found[i] = store.docs[internalId]
   }
 
   return found
 }
 
-export async function getAll(store: DocumentsStore): Promise<Record<string, Document>> {
-  return store.docs as Record<string, Document>
+export async function getAll(store: DocumentsStore): Promise<Record<InternalDocumentID, Document>> {
+  return store.docs as Record<InternalDocumentID, Document>;
 }
 
-export async function store(store: DocumentsStore, id: string, doc: Document): Promise<boolean> {
-  if (typeof store.docs[id] !== 'undefined') {
+export async function store(store: DocumentsStore, id: DocumentID, doc: Document): Promise<boolean> {
+  const internalId = getInternalDocumentId(store.sharedInternalDocumentStore, id);
+
+  if (typeof store.docs[internalId] !== 'undefined') {
     return false
   }
 
-  store.docs[id] = doc
+  store.docs[internalId] = doc
   store.count++
 
   return true
 }
 
-export async function remove(store: DocumentsStore, id: string): Promise<boolean> {
-  if (typeof store.docs[id] === 'undefined') {
+export async function remove(store: DocumentsStore, id: DocumentID): Promise<boolean> {
+  const internalId = getInternalDocumentId(store.sharedInternalDocumentStore, id);
+
+  if (typeof store.docs[internalId] === 'undefined') {
     return false
   }
 
-  delete store.docs[id]
+  delete store.docs[internalId]
   store.count--
 
   return true
```

Comment:
we should remove also from `sharedInternalDocumentStore`, right?

Reply:
If we just remove it from `sharedInternalDocumentStore.idToInternalId`, it's fast enough.

If we try to remove it from `sharedInternalDocumentStore.internalIdToId`, it will be slower as `sorter`.

What we can do is perform a cleanup of `internalIdToId` on serialize.

What do you prefer?

Reply:
No sorry, we should implement it.
For now, we don't care about the performance during the remove.

Reply:
> For now, we don't care about the performance during the remove.

Agreed. I'd dedicate a separate PR to this